R/component-LSM.R

```diff
@@ -0,0 +1,292 @@
+          edges = edge.list.rows[[dd]]
+        )
+        ldens.pos.orig <- dnorm(
+          x = latent.space.pos.hold[dd, ],
+          mean = 0,
+          sd = sqrt(latent.space.pos.v),
+          log = TRUE
+        )
+        log.dens.orig <- llike.orig + sum(ldens.pos.orig)
+        
+        # accepts or rejects proposal by comparing ratio to uniform(0, 1)
+        if (log.dens.prop - log.dens.orig > -rexp(1)) {
+          latent.space.pos.hold[dd, ] <- latent.space.pos.prop[dd, ]
+        } else {
+          # what does this do? latent.space.pos.prop is never used again
+          latent.space.pos.prop[dd, ] <- latent.space.pos.hold[dd, ]
+        }
+      }
+      
+      ## Procrustean transformation using random start as a target matrix
+      latent.space.pos.hold <- private$post.procrustean(
+        latent.space.pos.hold,
+        self$latent.space.target
+      )
+      rownames(latent.space.pos.hold) <- node.names
+      
+      # store accepted proposal/previously accepted draw as new draw
+      self$latent.space.pos <- latent.space.pos.hold
+      
+      # Gibbs update draw from posterior of latent space position variance
+      a.factor <- nrow(latent.space.pos) * ncol(latent.space.pos) / 2
+      b.factor <- sum(latent.space.pos^2) / 2
+      g.draw <- rgamma(n = 1,
+                       shape = a.factor + latent.space.pos.v.ab[1],
+                       rate = b.factor + latent.space.pos.v.ab[2]
+      )
+      self$latent.space.pos.v <- 1 / g.draw # stores in field
+    },
+    
+    # TODO: Determine which variables should be output (ls.pos, v)
+    create.output.list = function(total.draws) {
+      init <- replicate(total.draws, matrix(NA, self$dimension, private$n.nodes))
+      init.dimperm <- aperm(init, perm = 3:1)
+      return(list(
+        latent.space.pos = init.dimperm,
+        latent.space.pos.v = rep(NA, total.draws)
+      ))
+    },
+    
+    
+    update.output.list = function(gibbs.output.list, draw) {
+      gibbs.output.list$component_output$latent.space.pos[draw] = self$latent.space.pos
+      gibbs.output.list$component_output$latent.space.pos.v[draw] = self$latent.space.pos.v
+    },
+    
+    get.mean.component = function(gibbs.output.list) {
+      return(LSMParams$new(
+        dimension = self$dimension,
+        latent.space.pos = apply(gibbs.output.list$component_output$latent.space.pos, c(2, 3), mean),
+        latent.space.pos.v = mean(gibbs.output.list$component_output$latent.space.pos.v),
+        latent.space.target = self$latent.space.target,
+        inverted.model = self$inverted.model
+      ))
+    },
+    
+    value = function() {
+      "Returns latent distance between nodes multiplied by multiplicative factor."
+      val <- private$mult.factor *
+        edge.list.distance(self$latent.space.pos, private$edge.list)
+      return(val)
+    },
+    
+    # QUESTION: Do we want these functions here or should there be a separate
```

Comment:
I think we might want these here eventually.  Eventually we should probably have a separate top level summary function that uses the standard summary function.